cli/src/commands/bookmark/list.rs

```diff
@@ -270,3 +291,470 @@ struct RefListItem {
+    fn sort_and_snapshot(
+        items: &mut [RefListItem],
+        sort_keys: &[SortKey],
+        f: impl Fn(CommitId) -> Option<Arc<backend::Commit>>,
+    ) -> String {
+        let f = &f;
+        sort(items, sort_keys, f);
+        items
+            .iter()
+            .map(|item| {
+                sort_keys
+                    .iter()
+                    .map(|key| match key {
+                        SortKey::Name | SortKey::NameDesc => item.primary.name().to_owned(),
+                        SortKey::AuthorName | SortKey::AuthorNameDesc => {
+                            commit_for_ref_list_item(item, f)
+                                .map(|commit| format!("author: {}", commit.author.name))
+                                .unwrap_or_default()
+                        }
+                        SortKey::AuthorEmail | SortKey::AuthorEmailDesc => {
+                            commit_for_ref_list_item(item, f)
+                                .map(|commit| format!("author.email: {}", commit.author.email))
+                                .unwrap_or_default()
+                        }
+                        SortKey::AuthorDate | SortKey::AuthorDateDesc => {
+                            commit_for_ref_list_item(item, f)
+                                .map(|commit| {
+                                    format!(
+                                        "author.timestamp: {}",
+                                        commit.author.timestamp.timestamp.0
+                                    )
+                                })
+                                .unwrap_or_default()
+                        }
+                        SortKey::CommitterName | SortKey::CommitterNameDesc => {
+                            commit_for_ref_list_item(item, f)
+                                .map(|commit| format!("committer: {}", commit.committer.name))
+                                .unwrap_or_default()
+                        }
+                        SortKey::CommitterEmail | SortKey::CommitterEmailDesc => {
+                            commit_for_ref_list_item(item, f)
+                                .map(|commit| {
+                                    format!("committer.email: {}", commit.committer.email)
+                                })
+                                .unwrap_or_default()
+                        }
+                        SortKey::CommitterDate | SortKey::CommitterDateDesc => {
+                            commit_for_ref_list_item(item, f)
+                                .map(|commit| {
+                                    format!(
+                                        "committer.timestamp: {}",
+                                        commit.committer.timestamp.timestamp.0
+                                    )
+                                })
+                                .unwrap_or_default()
+                        }
+                    })
+                    .join(", ")
+            })
```

Comment:
nit: isn't it simpler to just dump all information unconditionally?

Reply:
Yes, of course. My concern is that not all fields are relevant for a particular test and just add noise.